### ISSUE TITLE
Crash of OpenLane in Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by hpretl** [The-OpenROAD-Project/OpenLane#1550]
### Description

Running `flow.tcl` on Ubuntu 22.04 LTS causes the following crash (Ubuntu 20.04 LTS works fine):

```
[ERROR]: during executing: "openroad -exit -no_init -python /foss/tools/openlane/2022.12/scripts/odbpy/lefutil.py get_metal_layers -o /foss/designs/spm/runs/foo/tmp/layers.list /foss/pdks/sky130A/libs.ref/sky130_fd_sc_hd/techlef/sky130_fd_sc_hd__nom.tlef"
[ERROR]: Exit code: 1
[ERROR]: Last 10 lines:
OpenROAD v2.0-5825-g7c85c1403 
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
Traceback (most recent call last):
  File "/foss/tools/openlane/2022.12/scripts/odbpy/lefutil.py", line 15, in <module>
    import click
ModuleNotFoundError: No module named 'click'

no such variable
    (read trace on "::env(CURRENT_DEF)")
    invoked from within
"lappend arg_list -def_path $::env(CURRENT_DEF)"
    (procedure "save_final_views" line 18)
    invoked from within
"save_final_views"
    (procedure "flow_fail" line 5)
    invoked from within
"flow_fail"
    (procedure "try_catch" line 17)
    invoked from within
"try_catch $::env(OPENROAD_BIN) -exit -no_init -python $::env(SCRIPTS_DIR)/odbpy/lefutil.py get_metal_layers -o $::env(TMP_DIR)/layers.list $arg_values..."
    (procedure "prep_lefs" line 36)
    invoked from within
"prep_lefs -tech_lef $::env(TECH_LEF) -corner nom -env_var MERGED_LEF"
    (procedure "prep" line 321)
    invoked from within
"prep {*}$args"
    (procedure "run_non_interactive_mode" line 12)
    invoked from within
"run_non_interactive_mode {*}$argv"
    invoked from within
"if { [info exists flags_map(-interactive)] || [info exists flags_map(-it)] } {
    if { [info exists arg_values(-file)] } {
        run_file [file nor..."
    (file "/foss/tools/openlane/2022.12/flow.tcl" line 401)
```

### Expected Behavior

Running without crash.

### Environment report

```shell
Kernel: Linux v5.15.49-linuxkit
Distribution: ubuntu 22.04
Python: v3.10.6 (OK)
Critical Alert: No Docker or Docker-compatible container engine was found.
OpenLane Git Version: b2bd5f68d29aa4bde1e7f7394745568e09e73e54
pip: INSTALLED
python-venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

b2bd5f6 2022-12-03T15:08:33+02:00 [BOT] Update PDK (#1532) - Openlane Bot -  (HEAD, tag: 2022.12.04)
7bb988c 2022-12-01T21:31:26+00:00 [BOT] Update magic (#1444) - Openlane Bot -  (tag: 2022.12.02)
95c7cd1 2022-12-01T17:23:19+02:00 magic updates (#1540) - Kareem Farid -  ()
---
Git Remotes

origin	https://github.com/The-OpenROAD-Project/OpenLane (fetch)
origin	https://github.com/The-OpenROAD-Project/OpenLane (push)
```

### Reproduction material

OpenLane version: 2022.12.04

### Relevant log output

```shell
default@139cc65ab3aa:/foss/designs$ flow.tcl -design spm -tag foo -overwrite
OpenLane b2bd5f68d29aa4bde1e7f7394745568e09e73e54
All rights reserved. (c) 2020-2022 Efabless Corporation and contributors.
Available under the Apache License, version 2.0. See the LICENSE file for more details.

[INFO]: Using configuration in 'spm/config.json'...
[INFO]: PDK Root: /foss/pdks
[INFO]: Process Design Kit: sky130A
[INFO]: Standard Cell Library: sky130_fd_sc_hd
[INFO]: Optimization Standard Cell Library: sky130_fd_sc_hd
[INFO]: Run Directory: /foss/designs/spm/runs/foo
[INFO]: Removing existing /foss/designs/spm/runs/foo...
[INFO]: Preparing LEF files for the nom corner...
[ERROR]: during executing: "openroad -exit -no_init -python /foss/tools/openlane/2022.12/scripts/odbpy/lefutil.py get_metal_layers -o /foss/designs/spm/runs/foo/tmp/layers.list /foss/pdks/sky130A/libs.ref/sky130_fd_sc_hd/techlef/sky130_fd_sc_hd__nom.tlef"
[ERROR]: Exit code: 1
[ERROR]: Last 10 lines:
OpenROAD v2.0-5825-g7c85c1403 
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
Traceback (most recent call last):
  File "/foss/tools/openlane/2022.12/scripts/odbpy/lefutil.py", line 15, in <module>
    import click
ModuleNotFoundError: No module named 'click'

no such variable
    (read trace on "::env(CURRENT_DEF)")
    invoked from within
"lappend arg_list -def_path $::env(CURRENT_DEF)"
    (procedure "save_final_views" line 18)
    invoked from within
"save_final_views"
    (procedure "flow_fail" line 5)
    invoked from within
"flow_fail"
    (procedure "try_catch" line 17)
    invoked from within
"try_catch $::env(OPENROAD_BIN) -exit -no_init -python $::env(SCRIPTS_DIR)/odbpy/lefutil.py get_metal_layers -o $::env(TMP_DIR)/layers.list $arg_values..."
    (procedure "prep_lefs" line 36)
    invoked from within
"prep_lefs -tech_lef $::env(TECH_LEF) -corner nom -env_var MERGED_LEF"
    (procedure "prep" line 321)
    invoked from within
"prep {*}$args"
    (procedure "run_non_interactive_mode" line 12)
    invoked from within
"run_non_interactive_mode {*}$argv"
    invoked from within
"if { [info exists flags_map(-interactive)] || [info exists flags_map(-it)] } {
    if { [info exists arg_values(-file)] } {
        run_file [file nor..."
    (file "/foss/tools/openlane/2022.12/flow.tcl" line 401)
```

**Comment by donn** [comment]
Is there any reason you're not using the container?

```
Line 10: ModuleNotFoundError: No module named 'click'
```

There's your problem at any rate: the Python dependencies for OpenLane have not been installed in this particular Ubuntu 22.04 environment. From the root of the repository, invoke `python3 -m pip install -r ./requirements.txt`. However, it is likely you will run into more issues using OpenLane un-containerized at the moment.

**Comment by hpretl** [comment]
Yes, because we install a whole set of tools into one container image (https://github.com/iic-jku/iic-osic-tools).

The requirements are fulfilled (just `install` is missing, but this is not reported in the fail), and so far, everything worked when we do a build (CentOS 8, RockyLinux 9, Ubuntu 20.04 LTS). Just Ubuntu 22.04 does not work for whatever reason.

**Comment by hpretl** [comment]
That works:

```PYTHONPATH=$(python -c "import sys; print(':'.join(x for x in sys.path if x))") && export PYTHONPATH ```

However, it would be good to fix the root cause (in Ubuntu 20.04 LTS and RockyLinux 8 setting the `PYTHONPATH` was not needed, only in Ubuntu 22.04 LTS).
